Sources/PathKit.swift

```diff
@@ -250,6 +256,60 @@ extension Path {
 
     return pathExtension
   }
+
+  /// Returns the relative path necessary to go from `base` to `self`.
+  ///
+  /// Both paths must be absolute or relative paths.
+  /// - throws: Throws an error when the path types do not match, or when `base` has so many parent path components
+  ///           that it refers to an unknown parent directory.
+  public func relativePath(from base: Path) throws -> Path {
+    enum PathArgumentError: Error {
```

Comment:
As this is specific for relative paths, should we name it as such?

```suggestion
    enum RelativePathError: Error {
```